openmetadata-service/src/main/java/org/openmetadata/service/search/security/RBACConditionEvaluator.java

```diff
@@ -43,8 +46,9 @@ public OMQueryBuilder evaluateConditions(SubjectContext subjectContext) {
         it.hasNext(); ) {
       SubjectContext.PolicyContext context = it.next();
       for (CompiledRule rule : context.getRules()) {
-        if (rule.getCondition() != null
-            && rule.getOperations().stream().anyMatch(SEARCH_RELEVANT_OPS::contains)) {
+        if ((rule.getCondition() != null
+                && rule.getOperations().stream().anyMatch(SEARCH_RELEVANT_OPS::contains))
```

Comment:
@harshach here i cannot find another way, since the domain only access role  uses All resources and on all metadata operation, any better ideas ?

Reply:
@mohityadav766 not sure I follow, what case are we missing?

Reply:
@harshach what i mean was, that domain restrictions apply to all type of metadataoperation, while the search to es mappings only is applied when VIEW_BASIC or VIEW_ALL operation is performed, so i had to manually look for DomainAccessRule and apply condition bypassing or that is below : 
```if ((rule.getCondition() != null
                && rule.getOperations().stream().anyMatch(SEARCH_RELEVANT_OPS::contains))
            || rule.getName().equalsIgnoreCase(DOMAIN_ONLY_ACCESS_RULE))```
            
 lmk if this makes sense

Reply:
@mohityadav766 the domain only access rule should come in from our policies while we are looping on them.
The goal of this check is when to enforce a rule on search related queries. The reason to do this is search related access is only related for view or view all. If someone has edit only rule that by default allows them to view atleast.